32 unit tests card assoc classes

stephensmeal opened the pull request:
Requesting peer review of the following class unit tests to be merged into the integ branch:
Card - test_card.py
CardType - test_cardtype.py
Hand - test_hand.py
Deck - test_deck.py

stephensmeal committed:
Recommitting Unit Tests For Card Assoc Classes To Fix Branch Issue

Recommitting the unit tests for the Card associated classes because
of a previous issue with the branch.
Card - test_card.py
CardType - test_cardtype.py
Deck - test_deck.py
Hand - test_hand.py

patmay commented on test/unit/test_deck.py:
```diff
@@ -0,0 +1,80 @@
+
+
+def test_contains_card():
+    if test_deck1.contains_card(test_card1) is True:
```
Refactor when you have the time.

patmay commented on test/unit/test_cardtype.py:
```diff
@@ -0,0 +1,8 @@
+"""test_cardtype.py"""
```
This test is not really necessary, and might cause unneeded work if it breaks. The int value of the Enums isn't important, so testing isn't valuable.

patmay commented on test/unit/test_card.py:
```diff
@@ -0,0 +1,28 @@
+"""test_card.py"""
```
Because card doesn't have much functionality, I think only testing the test_str() function would have been sufficient.

patmay commented on test/unit/test_hand.py:
```diff
@@ -0,0 +1,46 @@
+
+def test_init():
+    assert type(test_hand) is Hand
+
```
It's not really necessary to do this type of test.

patmay reviewed (dismissed):
All tests pass. Tests are probably too thorough.

stephensmeal commented:
Addressed comments... refactored Assertions for better readability, removed CardType tests, I did not remove additional test methods but will keep that in mind when writing future tests to eliminate unnecessary efforts.

yo09975 commented:
Please remove your merge commit 41cd50f4, that should have been a rebase and not a merge.

stephensmeal committed:
Corrected Card Assoc Unit Tests For Better Readability

patmay reviewed (approved):
Code unchanged. Git history looks tidy.
